test/atc_test.coffee

```diff
@@ -6,202 +6,223 @@ chai.use require 'sinon-chai'
 Helper = require('hubot-test-helper')
 helper = new Helper('../src')
 
+Locks = require('../src/locks');
+
 expect = chai.expect
 assert = chai.assert
 
 describe 'hubot-atc', ->
-  room = null
 
   beforeEach ->
     @clock = sinon.useFakeTimers()
-    room = helper.createRoom()
+    @room = helper.createRoom()
+    @room.robot.brain.data.applications = []
+    @room.robot.brain.data.environments = {}
 
   afterEach ->
     @clock.restore()
+    @room.destroy()
 
   lastMessage = (room) ->
     lastMessageIndex = room.messages.length - 1
     room.messages[lastMessageIndex][1]
 
+  addApplication = (room, applicationName) ->
+    room.robot.brain.data.applications.push(applicationName);
+    room.robot.brain.data.environments[applicationName] = [];
+
+  addApplicationEnvironment = (room, applicationName, environmentName) ->
+    room.robot.brain.data.environments[applicationName].push(environmentName);
+
+  deployApplication = (room, userName, applicationName, environmentName, timePeriod = null, timeUnit = null) ->
+    locks = new Locks()
+    deploymentDetails = { applicationName: applicationName, environmentName: environmentName }
+    lockDuration = {value: timePeriod, unit: timeUnit}
+
+    locks.add(deploymentDetails, userName, "master", lockDuration)
+
+    room.robot.brain.data.locks = locks
+
   describe "applications", ->
     describe "adding", ->
       context "when there are none", ->
         it 'adds the application', ->
-          room.user.say "duncan", "hubot application add foo"
-          expect(room.robot.brain.data.applications).to.include "foo"
+          @room.user.say("duncan", "hubot application add foo").then =>
```

Comment:
TIL - CoffeeScript is nice here, as it implicitly returns the Promise for mocha to wait on :+1: